### DISCUSSION TITLE
Rendering map efficiently in SPA

### DISCUSSION THREAD

**Discussion by farznia** [maplibre/maplibre-gl-js#4603] [Q&A]
I have an issue, In order to prewarming map in my SPA, I've wrote a map Provider and store the map reference in the Contex. In pages that I don't want to show the map I set visibility:hidden so user can navigate between pages smoothly without waiting for map to be ready but this solution affect LCP of my page however I used dynamic import for loading map, Performance insight tab show multiple long tasks related to the map that execute before rendering longest content. Can you help me for better solution?

**Comment by HarelM** [comment]
I think your best bet is to show a placeholder image instead of the map.
Maybe you can use the csp build. IDK, I not much help when it comes to performance improvements...
